.github/workflows/run_mkosi.sh

```diff
@@ -9,9 +9,17 @@ EC=0
 TEMPFILE="$(mktemp)"
 trap "rm -f '$TEMPFILE'" EXIT
 
+# We need isc-dhcp-server to be installed for the networkd unit tests, but we don't want to
+# run it by default
+TEMP_EXTRA_TREE="$(mktemp --directory)"
+trap "rm -rf '$TEMP_EXTRA_TREE'" EXIT
```

Comment:
Unfortunately, this won't work (I mean, it doesn't matter in this case, since the machine gets scraped anyway), as you can have only one handler per signal in bash:

```
$ bash
$ trap 'echo 1' EXIT
$ trap 'echo 2' EXIT
$ exit
exit
2
```

So I guess moving them to a function should work

```bash
at_exit() {
    set +e
    [[ -n "${TEMPFILE:-}" ]] && rm -fr "$TEMPFILE"
    [[ -n "${TEMP_EXTRA_TREE:-}" ]] && rm -fr "$TEMP_EXTRA_TREE"
}

trap at_exit EXIT
```

or something along these lines.

Reply:
good catch, fixed by simply removing both dirs